### ISSUE TITLE
TS breakpoints not working in Heroes tutorial using platform-browser-dynamic

### ISSUE THREAD

**Issue by greglarson** [Microsoft/vscode-chrome-debug#300]
The chrome debug extension does not work with the latest version of the Angular 2 Heroes tutorial with Typescript breakpoints. Breakpoints are ignored with the "... sourcemap?" error. The main difference in the project appears to be moving from 

"angular2": "2.0.0-beta.13", to the new @angular packages, and main.ts is now using the @angular/platform-browser-dynamic' rather than angular2/platform/browser.

Nothing I do with source maps makes any difference in getting the breakpoints to work.

Has anyone else seen this or able to get Heroes to debug in typescript with the chrome extension?

**Comment by roblourens** [comment]
What does `platform-browser-dynamic` do differently? I'll try this out when I get a chance, thanks.

**Comment by greglarson** [comment]
One other piece of info:
When the app reloads on saving an edited ts file, the breakpoint is hit. The initial launch of Chrome (F5) or Restart on the debug toolbar does not hit the breakpoint.
Thx

**Comment by greglarson** [comment]
Also, the app breaks at this line (621) in zone.js when the app reloads on save:

// In MS Edge this throws
fetchPromise = global['fetch']();

**Comment by roblourens** [comment]
Yeah, because it's throwing an exception and you have 'break on exceptions' on

**Comment by gmhispagnol** [comment]
I reproduced a similar issue. It is related to dynamic code + sourcemaps:

1) If I put a breakpoint inside a .ts that is loaded dynamically, sometimes it hits the breakpoint, sometime not;
2) If I put an 'alert' right after loading the dynamic part, but before entering the code, it works after I dismiss the alert;
3) If I put the breakpoint directly in .js file and turn off sourcemaps, it always works.

So, it seems to be a timing issue with dynamic loaded sourcemaps.

**Comment by roblourens** [comment]
Yeah, that's expected. It will be a race between setting the breakpoint, and that code executing. If you refresh the page, it should hit the breakpoint consistently the second time.

**Comment by roblourens** [comment]
Can you open a new issue for this with more details? No, we can't do it the same way in Chrome, because 
- We usually can't the path of a script
- We can't start it in a paused state to set BPs before scripts run

**Comment by roblourens** [comment]
@greglarson This works for me - I just cloned the repo and was able to debug it with a normal launch config -

```
{
            "name": "Launch Chrome against localhost, with sourcemaps",
            "type": "chrome",
            "request": "launch",
            "url": "http://localhost:8000",
            "sourceMaps": true,
            "webRoot": "${workspaceRoot}"
}
```

**Comment by roblourens** [comment]
Can you try again with the latest extension and vscode versions?

**Comment by greglarson** [comment]
I tried again with the latest versions and it is working. Thanks!
